Taskfile.yml

```diff
@@ -0,0 +1,86 @@
+version: "3"
```

Comment:
To be added in a future PR to enable `clang-tidy` checks:
```
  clang-tidy:
    vars:
      FLAGS: "-p {{.G_WSAM_BUILD_DIR}}/compile_commands.json --config-file=src/clp_ffi_js/.clang-tidy --extra-arg='-I{{.G_EMSCIPRTEN_DIR}}/upstream/emscripten/system/include'"
    cmds:
      - |-
        find src/clp_ffi_js \
          -type f \
          \( -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" -o -iname "*.inc" \) \
          -print0 | \
            xargs -0 clang-tidy {{.FLAGS}}
```

Reply:
```yml
  clang-tidy:
    vars:
      FLAGS: >-
        -p {{.G_CLP_FFI_JS_BUILD_DIR}}/compile_commands.json
        --config-file=src/clp_ffi_js/.clang-tidy
    deps:
      - "clp-ffi-js"
      - "emsdk"
    sources:
      - "{{.TASKFILE}}"
      - "src/clp_ffi_js/.clang-tidy"
      - "src/clp_ffi_js/**/*.cpp"
      - "src/clp_ffi_js/**/*.h"
      - "src/clp_ffi_js/**/*.hpp"
    cmds:
      - |-
        EXTRA_ARGS=$("{{.G_EMSDK_DIR}}/upstream/emscripten/em++" --cflags \
          | tr ' ' '\n' \
          | sed 's/^/--extra-arg /' \
          | tr '\n' ' ')
        find src/clp_ffi_js \
          -type f \
          \( -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" \) \
          -print0 | \
            xargs -0 clang-tidy {{.FLAGS}} $EXTRA_ARGS
```

Reply:
(also need to mark `emsdk` as `run: "once"`)

Reply:
Could you help confirm my understanding:

If I understand correctly, here are the things we did to get clang-tidy working:
1. pass the emsdk `--cflags` to `clang-tidy`, which includes the sysroot and a few other compiler options that can be passed to `clang`.
2. mark include directories of our dependencies as `SYSTEM` in the `CMakeLists.txt` so that with the generated compile_commands.json `clang-tidy` will treat those as non-project code and does not report errors unless we explicitly asks it to.

Reply:
Yep